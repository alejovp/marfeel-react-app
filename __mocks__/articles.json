{
    "items": [
        {
            "id": "art-1",
            "img": "https://burst.shopifycdn.com/photos/business-flatlay-in-india.jpg?width=373&format=pjpg&exif=0&iptc=0",
            "title": "Aute sunt nulla nostrud exercitation amet voluptate Lorem cillum incididunt proident.",
            "description": "Excepteur laborum quis anim dolor dolore proident culpa mollit ea aliquip elit culpa. Non voluptate qui Lorem laborum ad ex ipsum excepteur incididunt esse. Aliquip sunt in est voluptate cillum pariatur. Magna tempor labore esse quis dolor esse voluptate. Labore anim aliquip sint ullamco consectetur ad exercitation eu proident."
        },
        {
            "id": "art-2",
            "img": "https://burst.shopifycdn.com/photos/woman-edits-images-on-her-laptop.jpg?width=373&format=pjpg&exif=0&iptc=0",
            "title": "Sint consectetur et aute qui qui reprehenderit incididunt incididunt sunt.",
            "description": "Pariatur mollit quis occaecat in qui aute aute. Qui aute adipisicing et ea sit veniam ut officia eu pariatur tempor deserunt. Commodo consectetur ipsum Lorem magna quis labore ullamco est. Enim est enim fugiat aliquip veniam non. Sunt sunt proident dolore qui ipsum veniam incididunt exercitation adipisicing ut ex."
        },
        {
            "id": "art-3",
            "img": "https://burst.shopifycdn.com/photos/a-blue-point-himalayan-sits-at-a-desk.jpg?width=373&format=pjpg&exif=0&iptc=0",
            "title": "Voluptate mollit elit ipsum eiusmod voluptate dolor pariatur tempor ex.",
            "description": "Amet aliqua eiusmod Lorem minim consectetur non esse sint non commodo veniam cillum. Anim labore labore excepteur dolor cupidatat incididunt dolor quis magna nulla nostrud est. Reprehenderit eiusmod dolor officia deserunt cupidatat non ipsum. Anim non irure amet exercitation aute duis laboris enim dolor incididunt irure aute. Qui aliquip incididunt reprehenderit occaecat aliqua adipisicing sint ex nostrud ad mollit."
        },
        {
            "id": "art-4",
            "img": "https://burst.shopifycdn.com/photos/design-plans-color-swatches.jpg?width=373&format=pjpg&exif=0&iptc=0",
            "title": "Do laborum dolore qui adipisicing occaecat aute culpa amet proident.",
            "description": "Commodo ad ut proident nisi nisi ut. Magna nostrud officia ipsum magna quis nisi sunt mollit do veniam ex. Ullamco velit consectetur nisi non mollit eiusmod ullamco. Eu duis in ipsum aliquip fugiat elit consectetur ad."
        },
        {
            "id": "art-5",
            "img": "https://burst.shopifycdn.com/photos/desk-setup-with-motivational-poster.jpg?width=373&format=pjpg&exif=0&iptc=0",
            "title": "Aute sunt nulla nostrud exercitation amet voluptate Lorem cillum incididunt proident.",
            "description": "Excepteur laborum quis anim dolor dolore proident culpa mollit ea aliquip elit culpa. Non voluptate qui Lorem laborum ad ex ipsum excepteur incididunt esse. Aliquip sunt in est voluptate cillum pariatur. Magna tempor labore esse quis dolor esse voluptate. Labore anim aliquip sint ullamco consectetur ad exercitation eu proident."
        },
        {
            "id": "art-6",
            "img": "",
            "title": "Aute sunt nulla nostrud exercitation amet voluptate Lorem cillum incididunt proident.",
            "description": "Excepteur laborum quis anim dolor dolore proident culpa mollit ea aliquip elit culpa. Non voluptate qui Lorem laborum ad ex ipsum excepteur incididunt esse. Aliquip sunt in est voluptate cillum pariatur. Magna tempor labore esse quis dolor esse voluptate. Labore anim aliquip sint ullamco consectetur ad exercitation eu proident."
        },
        {
            "id": "art-7",
            "img": "https://burst.shopifycdn.com/photos/woman-edits-images-on-her-laptop.jpg?width=373&format=pjpg&exif=0&iptc=0",
            "title": "Sint consectetur et aute qui qui reprehenderit incididunt incididunt sunt.",
            "description": "Pariatur mollit quis occaecat in qui aute aute. Qui aute adipisicing et ea sit veniam ut officia eu pariatur tempor deserunt. Commodo consectetur ipsum Lorem magna quis labore ullamco est. Enim est enim fugiat aliquip veniam non. Sunt sunt proident dolore qui ipsum veniam incididunt exercitation adipisicing ut ex."
        },
        {
            "id": "art-8",
            "img": "https://burst.shopifycdn.com/photos/a-blue-point-himalayan-sits-at-a-desk.jpg?width=373&format=pjpg&exif=0&iptc=0",
            "title": "Voluptate mollit elit ipsum eiusmod voluptate dolor pariatur tempor ex.",
            "description": "Amet aliqua eiusmod Lorem minim consectetur non esse sint non commodo veniam cillum. Anim labore labore excepteur dolor cupidatat incididunt dolor quis magna nulla nostrud est. Reprehenderit eiusmod dolor officia deserunt cupidatat non ipsum. Anim non irure amet exercitation aute duis laboris enim dolor incididunt irure aute. Qui aliquip incididunt reprehenderit occaecat aliqua adipisicing sint ex nostrud ad mollit."
        },
        {
            "id": "art-9",
            "img": "https://burst.shopifycdn.com/photos/design-plans-color-swatches.jpg?width=373&format=pjpg&exif=0&iptc=0",
            "title": "Do laborum dolore qui adipisicing occaecat aute culpa amet proident.",
            "description": "Commodo ad ut proident nisi nisi ut. Magna nostrud officia ipsum magna quis nisi sunt mollit do veniam ex. Ullamco velit consectetur nisi non mollit eiusmod ullamco. Eu duis in ipsum aliquip fugiat elit consectetur ad."
        },
        {
            "id": "art-10",
            "img": "https://burst.shopifycdn.com/photos/desk-setup-with-motivational-poster.jpg?width=373&format=pjpg&exif=0&iptc=0",
            "title": "Aute sunt nulla nostrud exercitation amet voluptate Lorem cillum incididunt proident.",
            "description": "Excepteur laborum quis anim dolor dolore proident culpa mollit ea aliquip elit culpa. Non voluptate qui Lorem laborum ad ex ipsum excepteur incididunt esse. Aliquip sunt in est voluptate cillum pariatur. Magna tempor labore esse quis dolor esse voluptate. Labore anim aliquip sint ullamco consectetur ad exercitation eu proident."
        },
        {
            "id": "art-11",
            "img": "https://burst.shopifycdn.com/photos/business-flatlay-in-india.jpg?width=373&format=pjpg&exif=0&iptc=0",
            "title": "Aute sunt nulla nostrud exercitation amet voluptate Lorem cillum incididunt proident.",
            "description": "Excepteur laborum quis anim dolor dolore proident culpa mollit ea aliquip elit culpa. Non voluptate qui Lorem laborum ad ex ipsum excepteur incididunt esse. Aliquip sunt in est voluptate cillum pariatur. Magna tempor labore esse quis dolor esse voluptate. Labore anim aliquip sint ullamco consectetur ad exercitation eu proident."
        },
        {
            "id": "art-12",
            "img": "https://burst.shopifycdn.com/photos/woman-edits-images-on-her-laptop.jpg?width=373&format=pjpg&exif=0&iptc=0",
            "title": "Sint consectetur et aute qui qui reprehenderit incididunt incididunt sunt.",
            "description": "Pariatur mollit quis occaecat in qui aute aute. Qui aute adipisicing et ea sit veniam ut officia eu pariatur tempor deserunt. Commodo consectetur ipsum Lorem magna quis labore ullamco est. Enim est enim fugiat aliquip veniam non. Sunt sunt proident dolore qui ipsum veniam incididunt exercitation adipisicing ut ex."
        },
        {
            "id": "art-13",
            "img": "https://burst.shopifycdn.com/photos/a-blue-point-himalayan-sits-at-a-desk.jpg?width=373&format=pjpg&exif=0&iptc=0",
            "title": "Voluptate mollit elit ipsum eiusmod voluptate dolor pariatur tempor ex.",
            "description": "Amet aliqua eiusmod Lorem minim consectetur non esse sint non commodo veniam cillum. Anim labore labore excepteur dolor cupidatat incididunt dolor quis magna nulla nostrud est. Reprehenderit eiusmod dolor officia deserunt cupidatat non ipsum. Anim non irure amet exercitation aute duis laboris enim dolor incididunt irure aute. Qui aliquip incididunt reprehenderit occaecat aliqua adipisicing sint ex nostrud ad mollit."
        },
        {
            "id": "art-14",
            "img": "https://burst.shopifycdn.com/photos/design-plans-color-swatches.jpg?width=373&format=pjpg&exif=0&iptc=0",
            "title": "Do laborum dolore qui adipisicing occaecat aute culpa amet proident.",
            "description": "Commodo ad ut proident nisi nisi ut. Magna nostrud officia ipsum magna quis nisi sunt mollit do veniam ex. Ullamco velit consectetur nisi non mollit eiusmod ullamco. Eu duis in ipsum aliquip fugiat elit consectetur ad."
        },
        {
            "id": "art-15",
            "img": "https://burst.shopifycdn.com/photos/desk-setup-with-motivational-poster.jpg?width=373&format=pjpg&exif=0&iptc=0",
            "title": "Aute sunt nulla nostrud exercitation amet voluptate Lorem cillum incididunt proident.",
            "description": "Excepteur laborum quis anim dolor dolore proident culpa mollit ea aliquip elit culpa. Non voluptate qui Lorem laborum ad ex ipsum excepteur incididunt esse. Aliquip sunt in est voluptate cillum pariatur. Magna tempor labore esse quis dolor esse voluptate. Labore anim aliquip sint ullamco consectetur ad exercitation eu proident."
        }
    ] 
}