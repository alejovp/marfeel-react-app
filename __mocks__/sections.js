export const sectionsMock = [
    {
        id: 'sec-1',
        text: 'SECTION 1',
        path: '/section-1'
    },
    {
        id: 'sec-2',
        text: 'SECTION 2',
        path: '/section-2'
    },
    {
        id: 'sec-3',
        text: 'SECTION 3',
        path: '/section-3'
    },
    {
        id: 'sec-4',
        text: 'SECTION 4',
        path: '/section-4'
    },
    {
        id: 'sec-5',
        text: 'SECTION 5',
        path: '/section-5'
    },
    {
        id: 'sec-6',
        text: 'SECTION 6',
        path: '/section-6'
    },
    {
        id: 'sec-7',
        text: 'SECTION 7',
        path: '/section-7'
    },
    {
        id: 'sec-8',
        text: 'SECTION 8',
        path: '/section-8'
    },
    {
        id: 'sec-9',
        text: 'SECTION 9',
        path: '/section-9'
    },
    {
        id: 'sec-10',
        text: 'SECTION 10',
        path: '/section-10'
    }
];
