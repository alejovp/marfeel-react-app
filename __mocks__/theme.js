export const themeMock = {
    tipography: {
        primaryFontColor: 'black',
        fontFamily: 'Helvetica, sans-serif'
    },
    header: {
        padding: 15
    },
    sectionMenu: {
        color: 'orange',
        textSize: 1.1 //em
    },
    background: {
        color: 'gray'
    },
    burgerButton: {
        color: 'red'
    }
};
