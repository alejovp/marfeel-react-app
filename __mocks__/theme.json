{
    "tipography": {
        "primaryFontColor": "black",
        "fontFamily": "Helvetica, sans-serif"
    },
    "header": {
        "padding": 15
    },
    "sectionMenu": {
        "color": "orange",
        "textSize": 1.1
    },
    "background": {
        "color": "gray"
    },
    "burgerButton": {
        "color": "red"
    },
    "logo": {
        "image": "https://www.marfeel.com/public/images/landing/header/logo-marfeel.svg",
        "size": 36
    }
}