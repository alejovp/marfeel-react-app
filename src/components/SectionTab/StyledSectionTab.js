import styled from 'styled-components';

const getBorderBottom = ({ theme, isActive }) => {
    const { tipography, sectionMenu } = theme;
    const defaultColor = tipography && tipography.primaryFontColor;
    let borderColor;
    if (isActive && sectionMenu && sectionMenu.color) {
        borderColor = sectionMenu.color;
    } else if (isActive) {
        borderColor = defaultColor;
    } else {
        borderColor = 'transparent';
    }
    return `4px solid ${borderColor}`;
};

export const StyledSectionTab = styled.li`
    list-style: none;
    padding: 8px 10px;
    text-align: center;
    cursor: pointer;
    color: ${({ theme }) => theme.sectionMenu && theme.sectionMenu.color};
    border-bottom: ${getBorderBottom};

    &:hover {
        opacity: 0.8
    }
`;

StyledSectionTab.displayName = 'StyledSectionTab';
