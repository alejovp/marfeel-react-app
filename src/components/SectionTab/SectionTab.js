import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { StyledSectionTab } from './StyledSectionTab';

export const SectionTab = ({ section = {}, onClick, currentSection }) => {

    const onClickHandler = useCallback(
        () => onClick(section.id), [onClick]
    );

    if (!section.id || !section.text) {
        return null;
    }

    const isActive = section.id === currentSection;

    return (
        <StyledSectionTab onClick={onClickHandler}
                          isActive={isActive}>
            <a>{section.text}</a>
        </StyledSectionTab>
    );
};

SectionTab.propTypes = {
    section: PropTypes.object.isRequired,
    onClick: PropTypes.func.isRequired,
    currentSection: PropTypes.string
};
