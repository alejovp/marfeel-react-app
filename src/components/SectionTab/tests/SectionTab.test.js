import React from 'react';
import { SectionTab } from '../SectionTab';
import { themeMock } from '../../../../__mocks__/theme';
import { shallowWithTheme } from '../../../../scripts/jest/setup';
import { sectionsMock } from '../../../../__mocks__/sections';
import { StyledSectionTab } from '../StyledSectionTab';


describe('<SectionTab />', () => {

    describe('shape', () => {
        const renderedComponent = shallowWithTheme(
            <SectionTab section={sectionsMock[1]}
                        onClick={jest.fn()} />, themeMock);

        it('will render correctly', () => {
            expect(renderedComponent).toMatchSnapshot();
        });

        it('will render a StyledSectionTab component', () => {
            expect(renderedComponent.find(StyledSectionTab)).toHaveLength(1);
        });

        it('will render an anchor elem', () => {
            expect(renderedComponent.find('a')).toHaveLength(1);
        });

    });

    describe('props', () => {

        it('wont render anything if section prop have no id or text', () => {
            const renderedComponent = shallowWithTheme(
                <SectionTab section={{ text: '' }}
                            onClick={jest.fn()} />, themeMock);
            expect(renderedComponent.type()).toBe(null);
        });

        it('will trigger the onClick callback with the section id', () => {
            const onClickMock = jest.fn();
            const renderedComponent = shallowWithTheme(
                <SectionTab section={sectionsMock[1]}
                            onClick={onClickMock}
                            currentSection={sectionsMock[2].id} />, themeMock);
            renderedComponent.simulate('click');
            expect(onClickMock).toBeCalledWith(sectionsMock[1].id);
        });

    });

});
