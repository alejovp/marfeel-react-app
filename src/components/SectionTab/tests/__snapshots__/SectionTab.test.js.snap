// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionTab /> shape will render correctly 1`] = `
<StyledSectionTab
  isActive={false}
  onClick={[Function]}
>
  <a>
    SECTION 2
  </a>
</StyledSectionTab>
`;
